testsuite/cluster-test/src/experiments/performance_benchmark.rs

```diff
@@ -76,47 +67,46 @@ impl PerformanceBenchmarkParams {
 impl ExperimentParam for PerformanceBenchmarkParams {
     type E = PerformanceBenchmark;
     fn build(self, cluster: &Cluster) -> Self::E {
-        if self.is_fullnode {
-            let num_nodes = cluster.fullnode_instances().len();
-            let nodes_down = (num_nodes * self.percent_nodes_down) / 100;
-            let (down_instances, up_instances) = cluster.split_n_fullnodes_random(nodes_down);
-            Self::E {
-                down_instances: down_instances.into_fullnode_instances(),
-                up_instances: up_instances.into_fullnode_instances(),
-                percent_nodes_down: self.percent_nodes_down,
-                duration: Duration::from_secs(self.duration),
-                trace: self.trace,
-            }
-        } else {
-            let num_nodes = cluster.validator_instances().len();
-            let nodes_down = (num_nodes * self.percent_nodes_down) / 100;
-            let (down_instances, up_instances) = cluster.split_n_validators_random(nodes_down);
-            Self::E {
-                down_instances: down_instances.into_validator_instances(),
-                up_instances: up_instances.into_validator_instances(),
-                percent_nodes_down: self.percent_nodes_down,
-                duration: Duration::from_secs(self.duration),
-                trace: self.trace,
-            }
+        Self::E {
+            cluster: cluster.clone(),
+            percent_nodes_down: self.percent_nodes_down,
+            duration: Duration::from_secs(self.duration),
+            trace: self.trace,
         }
     }
 }
 
 #[async_trait]
 impl Experiment for PerformanceBenchmark {
     fn affected_validators(&self) -> HashSet<String> {
-        instance::instancelist_to_set(&self.down_instances)
+        instance::instancelist_to_set(self.cluster.validator_instances())
```

Comment:
Why just returning all validator_instances? This would basically mean health checks does not work for this experiment.

This is probably why you do need `down_instances` initialized from params(and therefore pass global_emit_config into ExperimentParam::build). Otherwise you just don't have access to it from functions like this one

Reply:
> This would basically mean health checks does not work for this experiment.

Not really, we are still checking that every validator is healthy before passing the experiment.

Now that we will have both validators and fullnodes in the network, affected_validators is not very useful, we probably need to healthcheck every instance in the network.

Reply:
So health check runs during experiment as well. This is to check that our network is fault tolerant, and even bringing down 10% of validators, all other validators still pass the health check. This is why experiment exports `affected_validators`

Reply:
Updated experiment to stop validators instead of fullnodes.